### PR TITLE
Third week pull request.

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: app/src/main/java/com/strikelines/app/OsmandHelper.kt [hunk 1]
```diff
@@ -118,7 +155,19 @@ class OsmandHelper(private val app: Application) {
 	fun setNavDrawerLogoWithParams(uri:Uri, packageName:String, intent:String){
 		if (mIOsmAndAidlInterface != null) {
 			try {
-				mIOsmAndAidlInterface!!.setNavDrawerLogoWithParams(uri.toString(), packageName, intent)
+				mIOsmAndAidlInterface!!.setNavDrawerLogoWithParams(
+                    NavDrawerHeaderParams(uri.toString(), packageName, intent))
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/OsmandHelper.kt]: Spaces

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 2]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
                     Html.fromHtml(chart.description.clearGarbadge(), 0)
                 else Html.fromHtml(chart.description.clearGarbadge())
         GlideApp.with(details_image)
-                .load(chart.imageurl).placeholder(R.drawable.img_placeholder).into(details_image)
+            .load(chart.imageurl).placeholder(R.drawable.img_placeholder).into(details_image)
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Spaces

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 3]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
...
         } else {
             get_chart_btn.setOnClickListener { downloadFreeChart(chart.downloadurl) }
             get_chart_btn.text = getString(R.string.shop_details_btn_tag_free_chart)
         }
 
+
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
             details_scroll_view.setOnScrollChangeListener { v, scrollX, scrollY, oldScrollX, oldScrollY ->
-                Log.d("Scroll Change", "sY: $scrollY, olY: $oldScrollY")
+                Log.d("Scroll", "old: $oldScrollY, new: $scrollY")
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Remove development log

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 4]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
...
+                if (oldScrollY<scrollY&&details_back_btn.alpha!=0f) {
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Format code

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 5]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
...
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-
-                    val bounds = Rect()
-                    details_image.getHitRect(bounds)
-
-                    val scrollBounds = Rect(scrollX, scrollY,
-                            scrollX + details_scroll_view.width,
-                            scrollY + details_scroll_view.height)
+                }
+                else if (oldScrollY > scrollY && details_back_btn.alpha!=1f) {
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Format code

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 6]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
...
+        }
+    }
+
     private fun downloadFreeChart(downloadUrl: String) {
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
 
-            if (ActivityCompat.checkSelfPermission(this, android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
-                    == PackageManager.PERMISSION_GRANTED
+            if (ActivityCompat.checkSelfPermission(
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Previous code style was better

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 7]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
...
+                )
+                == PackageManager.PERMISSION_GRANTED
             ) {
                 DownloadFileAsync(downloadUrl).execute()
             } else {
-                requestPermissions(arrayOf(android.Manifest.permission.WRITE_EXTERNAL_STORAGE), DOWNLOAD_REQUEST_CODE)
+                requestPermissions(
+                    arrayOf(android.Manifest.permission.WRITE_EXTERNAL_STORAGE),
+                    DOWNLOAD_REQUEST_CODE
+                )
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Extra line feed?

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 8]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
...
         }
     }
 
+    private fun fadeOut(view: View, duration:Long = 500) {
+        view.apply {
+            animate()
+                .alpha(0f)
+                .setDuration(duration)
+                .setListener(null
+                )
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Do not move ')' on the next line

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 9]
```diff
@@ -72,85 +63,91 @@ class DetailedPurchaseChartScreen : AppCompatActivity() {
...
         }
 
     }
 
-    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Line feed?

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt [hunk 10]
```diff
@@ -79,15 +95,73 @@ class MainActivity : AppCompatActivity(), OsmandHelperListener {
...
+                installOsmadnDialog()
                 Toast.makeText(view.context, "OsmAnd Missing", Toast.LENGTH_SHORT).show()
             }
         }
+        fab.setOnTouchListener(
+            object : View.OnTouchListener {
+                override fun onTouch(v: View?, event: MotionEvent?): Boolean {
+                    when (event?.action) {
+                        MotionEvent.ACTION_UP -> {
+                            big_fab_icon.setColorFilter(
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: Too much line feeds.

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt [hunk 11]
```diff
@@ -114,32 +188,22 @@ class MainActivity : AppCompatActivity(), OsmandHelperListener {
...
-        super.onResume()
-        osmandHelper.listener = this
-        StrikeLinesApplication.listener = appListener
-    }
-
-    override fun onRestart() {
-        super.onRestart()
-        setupOsmand()
-    }
+    fun dismissLoader(){ loading_indicator.visibility = View.GONE }
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: Lack of line feeds

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt [hunk 12]
```diff
@@ -36,34 +44,42 @@ class MainActivity : AppCompatActivity(), OsmandHelperListener {
     private val listeners = mutableListOf<WeakReference<OsmandHelperListener>>()
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: Spaces

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt [hunk 13]
```diff
@@ -114,32 +188,22 @@ class MainActivity : AppCompatActivity(), OsmandHelperListener {
...
 
-        osmandHelper.listener = null
-        StrikeLinesApplication.listener = null
-    }
 
     override fun onOsmandConnectionStateChanged(connected: Boolean) {
+        if (connected) {
+            osmandHelper.registerForOsmandInitialization()
+            isOsmandConnected = true
+            if (isOsmandConnected && chartsDataIsReady) loading_indicator.visibility = View.GONE
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: Is it main thread?

File: app/src/main/java/com/strikelines/app/ui/MapsTabFragment.kt [hunk 14]
```diff
@@ -27,9 +28,9 @@ class MapsTabFragment : Fragment(), OsmandHelperListener, OnCheckedListener {
     private val viewAdapter = LocalItemsViewAdapter()
 
     override fun onCreateView(
-            inflater: LayoutInflater,
-            container: ViewGroup?,
-            savedInstanceState: Bundle?
+        inflater: LayoutInflater,
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/MapsTabFragment.kt]: Spaces

File: app/src/main/java/com/strikelines/app/OsmandHelper.kt (hunk 1 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/OsmandHelper.kt]: +

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt (hunk 2 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: +

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt (hunk 3 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: +

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt (hunk 4 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: +

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt (hunk 10 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: +

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt (hunk 5 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: +

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt (hunk 11 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: +

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt (hunk 12 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: +

**Pull request merged**